src/API/Endpoints/Reports/TopDonors.php

```diff
@@ -14,21 +14,48 @@ public function __construct() {
 		$this->endpoint = 'top-donors';
 	}
 
-	public function get_report($request) {
+	public function get_report( $request ) {
 
-		// Add caching logic here...
+		// Setup donor query args (get sanitized start/end date from request)
+		$args = [
+			'number'     => 25,
+			'paged'      => 1,
+			'orderby'    => 'purchase_value',
+			'order'      => 'DESC',
+			'start_date' => $request['start'],
+			'end_date'   => $request['end'],
+		];
 
-		return new \WP_REST_Response([
-			'data' => [
-				[
-					'type' => 'donor',
-					'name' => 'Name',
-					'count' => '4 Donations',
-					'total' => '$50.00',
-					'image' => 'image.png',
-					'email' => 'test@email.com'
-				],
+		// Get array of top 25 donors
+		$donors = new \Give_Donors_Query( $args );
+		$donors = $donors->get_donors();
+
+		// Populate $list with arrays in correct shape for frontend RESTList component
+		$list = [];
+		foreach ( $donors as $donor ) {
+
+			$avatar     = give_validate_gravatar( $donor->email ) ? get_avatar( $donor->email, 60 ) : null;
+			$total      = give_currency_filter( give_format_amount( $donor->purchase_value, array( 'sanitize' => false ) ), [ 'decode_currency' => true ] );
+			$url        = admin_url( 'edit.php?post_type=give_forms&page=give-donors&view=overview&id=' . absint( $donor->id ) );
+			$countLabel = $donor->purchase_count > 1 ? esc_html__( 'Donations', 'give' ) : esc_html__( 'Donation', 'give' );
+
+			$item = [
+				'type'  => 'donor',
+				'name'  => $donor->name,
+				'url'   => $url,
+				'count' => "{$donor->purchase_count} {$countLabel}",
+				'total' => $total,
+				'image' => $avatar,
+				'email' => $donor->email,
+			];
+			array_push( $list, $item );
```

Comment:
@henryholtgeerts can we use the shortened method.
I suggested same here: https://github.com/impress-org/givewp/pull/4420#discussion_r368825393

Reply:
Change made, ref: https://github.com/impress-org/givewp/pull/4430/commits/da20a7bd102ef43e220108b9f93effe1933c4068